pkg/identity/user.go

```diff
@@ -0,0 +1,242 @@
+package identity
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/scrypt"
+)
+
+type UserIdentity struct {
```

Comment:
high level comment: we should make sure UserIdentity is per user per community rather than tied to a node or habitat instance